torchao/dtypes/uintx/packed_linear_int8_dynamic_activation_intx_weight_layout.py

```diff
@@ -358,79 +418,35 @@ def make_packed_linear_int8_dynamic_activation_intx_weight_tensor(
     assert TORCH_VERSION_AT_LEAST_2_6, "Using PackedLinearInt8DynamicActivationIntxWeightLayout requires torch version > 2.6.0"
 
     layout = PackedLinearInt8DynamicActivationIntxWeightLayout(target=target)
-    if layout.target != Target.ATEN:
-        _check_torchao_ops_loaded()
-    else:
-        assert (
-            torch.backends.kleidiai.is_available()
-        ), "ATEN target requires torch.backends.kleidiai.is_available()"
-        assert data_dtype == torch.int4, "ATEN target only supports torch.int4"
-        assert zero_point is None, "ATEN target does not support zeros"
 
-    assert data_dtype in [getattr(torch, f"int{x}") for x in range(1, 9)]
-    qmin, qmax = _DTYPE_TO_QVALUE_BOUNDS[data_dtype]
     bit_width = _DTYPE_TO_BIT_WIDTH[data_dtype]
+    qmin, qmax = _DTYPE_TO_QVALUE_BOUNDS[data_dtype]
 
-    int_types = [torch.int8, torch.int16, torch.int32, torch.int64]
-
-    # Check int_data
-    assert int_data.device == torch.device("cpu")
-    assert int_data.dtype in int_types
     n, k = int_data.shape
-    if validate_inputs:
-        assert int_data.min().item() >= qmin
-        assert int_data.max().item() <= qmax
-
-    # Check scale
-    assert scale.device == torch.device("cpu")
-    if scale.dtype != torch.float32:
-        logging.info(f"scale has dtype {scale.dtype}, converting to torch.float32")
-        scale = scale.to(torch.float32)
     n_, groups_per_k = scale.shape
-    assert n_ == n
     assert k % groups_per_k == 0
     group_size = k // groups_per_k
-    if validate_inputs:
-        assert scale.min().item() > 0
 
-    if validate_inputs:
-        # Some targets round scales to bfloat16, give warning if scales are at higher precision
-        scale_is_rounded_to_bf16 = torch.allclose(
-            scale, scale.to(torch.bfloat16).to(torch.float32)
-        )
-        if not scale_is_rounded_to_bf16:
-            if layout.target == Target.ATEN and (group_size < k):
-                logging.warning(
-                    "When using Target.ATEN with group_size < k, scales will be rounded to bfloat16"
-                )
-            if layout.target in [Target.AUTO, Target.KLEIDIAI]:
-                logging.warning(
-                    "When using [Target.AUTO, Target.KLEIDIAI], scales will be rounded to bfloat16"
-                )
-
-    # Check zero_point
-    has_weight_zeros = zero_point is not None
-    if has_weight_zeros:
-        assert zero_point.device == torch.device("cpu")
-        assert zero_point.shape == scale.shape
-        assert zero_point.dtype in int_types
-        if validate_inputs:
-            assert zero_point.min().item() >= qmin
-            assert zero_point.max().item() <= qmax
+    has_weight_zeros = True
+    if zero_point is None:
+        has_weight_zeros = False
+    else:
+        zero_point_min = zero_point.min().item()
+        zero_point_max = zero_point.max().item()
+        if zero_point_min == 0 and zero_point_max == 0:
+            has_weight_zeros = False
```

Comment:
nit: might be better to have a util to check for zero_point being all zero and use it everywhere to keep consistent